### PR TITLE
feat(refs T35963): Add localisation and phases to statement meta

### PR DESCRIPTION
**Pull request by salisdemos**
This adds the phases and location data to the segments list meta data (new stn detail view)
Positioning and adjustments are quite likely necessary after all parts are merged together

**Ticket** https://demoseurope.youtrack.cloud/issue/DPLAN-3217/2-5-Integration-EWM-Part-2-Stellungnahmedetailansicht-aus-EWM-und-dplan-vereinen

### CODE REVIEW

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 1]
```diff
@@ -19,9 +19,10 @@
     </div>
 
     <div
-      class="u-mt-0_5"
+      class="u-mt-0_5 flex gap-2"
       data-dp-validate="statementMetaData">
-      <div class="inline-block u-1-of-2 align-top">
+      <div class="inline-block w-1/2 align-top">
+        <!-- @todo we have to add the logic here, that it could be an institution -->
```
Comment (salisdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue:25]: not part of the PR
```suggestion
```

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 2]
```diff
@@ -382,10 +472,26 @@ export default {
...
     syncAuthorAndSubmitter () {
       this.localStatement.attributes.submitName = this.localStatement.attributes.authorName
+    },
+
+    updateCounties (val) {
+      console.log(val, this.localStatement.attributes.counties)
+    },
+
+    updateMunicipalities (val) {
+      console.log(val, this.localStatement.attributes.municipaities)
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue:494]: Add update logic here, old logic is here client/js/components/statement/assessmentTable/DetailView/DetailView.vue:192

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    <dp-multiselect
+      v-else
+      :value="value"
+      @input="(changedValue) => $emit('change', changedValue)"
+      :id="name"
+      class="layout__item u-1-of-1 inline-block"
+      label="name"
+      multiple
+      :options="options"
+      track-by="id">
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue:31]: Please order correctly

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue [hunk 4]
```diff
@@ -0,0 +1,99 @@
+<template>
+  <div>
+    <dp-label
+      :text="label"
+      :for="name" />
+    <ul
+      v-if="editable"
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue:7]: I think the list should be shown if it is not editable and the multiselect if editable. Option names should be shown if they're defined and otherwise `-`.

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      :id="name"
+      class="o-list o-list--csv color--grey">
+      <template v-if="options.filter(opt => value.contains(opt))">
+        <li
+          v-for="option in options.filter(opt => value.contains(opt))"
+          :key="`county-${option.name}`"
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue:13]: I think `county` is too specific. Maybe `${name}-${option.name}`?

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue [hunk 6]
```diff
@@ -0,0 +1,99 @@
+<template>
+  <div>
+    <dp-label
+      :text="label"
+      :for="name" />
+    <ul
+      v-if="editable"
+      :id="name"
+      class="o-list o-list--csv color--grey">
+      <template v-if="options.filter(opt => value.contains(opt))">
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue:10]: Is method `contains` missing or do you mean `includes`?

**Review by gruenbergerdemos — CHANGES_REQUESTED** [review]

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 7]
```diff
@@ -185,6 +185,28 @@
...
+        <dp-select
+          id="statementProcedurePhase"
+          v-model="localStatement.attributes.phase"
+          class="u-mb-0_5"
+          :disabled="!editable"
+          :label="{
+            text: Translator.trans('procedure.public.phase')
+          }"
+          :options="availableExternalPhases"
+          @select="(val) => emitInput('submitType', val)" />
```
Comment (salisdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue:208]:
```suggestion
        <dp-select
          id="statementProcedureExternalPhase"
          v-model="localStatement.attributes.externalPhase"
          class="u-mb-0_5"
          :disabled="!editable"
          :label="{
            text: Translator.trans('procedure.public.phase')
          }"
          :options="availableExternalPhases"
          @select="(val) => emitInput('submitType', val)" />
```

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue (hunk 6 above)
Comment (sakutademos) [reply client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue]: yes, this method is deprecated. As the logic has been changed slightly (the prop `value` is an `Array` now instead of a `String`). here is a new code https://github.com/demos-europe/demosplan-core/commit/f8798491b7ed3830e73cbb3fe0394c83b7fb74e9#diff-4729d18dcde3cfb52dbf7c49d6a03e9e0b6734cd10b775a102fda74c9f7bd8b3R106-R111

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 8]
```diff
@@ -185,6 +185,28 @@
...
+
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue]: in progress ...

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 9]
```diff
@@ -185,6 +185,32 @@
...
+            id="statementProcedureExternalPhase"
+            v-model="localStatement.attributes.phase"
+            class="mb-3"
+            :disabled="!editable || !localStatement.attributes.isManual"
+            :label="{
+              text: Translator.trans('procedure.public.phase')
+            }"
+            :options="availableExternalPhases"
+            @select="(val) => emitInput('phase', val)" />
+        </template>
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue:212]: If these stay like that (basically the same code), I think we should merge them into one and dynamically change the parts that are different (`options`)

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+      selectedItems: this.value
+    }
+  },
+
+  computed: {
+    filterValueByOptions () {
+      // Filters the value array based on whether each item's name exists in the options array
+      return this.value.filter(item => {
+        return this.options.some(option => option.name === item.name)
+      })
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaMultiselect.vue]: done here https://github.com/demos-europe/demosplan-core/pull/2894/commits/f0280e06e5984a20c5a79814207e25e8ff949708

**Review by gruenbergerdemos — APPROVED** [review]
lgtm, thanks

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 11]
```diff
@@ -45,25 +45,7 @@
...
-          }"
-          @input="val => emitInput('initialOrganisationDepartmentName', val)" />
-        <dp-input
-          v-if="localStatement.attributes.represents"
-          id="statementRepresentation"
-          disabled
-          :label="{
-            text: Translator.trans('statement.representation.assessment')
-          }"
-          :value="localStatement.attributes.represents" />
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue:66]: I think this shouldn't be deleted

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 12]
```diff
@@ -45,25 +45,7 @@
...
-        <dp-input
-          v-if="hasPermission('field_statement_meta_orga_department_name') && !this.localStatement.attributes.isSubmittedByCitizen"
-          id="statementDepartmentName"
-          v-model="localStatement.attributes.initialOrganisationDepartmentName"
-          class="u-mb-0_5"
-          :disabled="isStatementManual ? false : !editable"
-          :label="{
-            text: Translator.trans('department')
-          }"
-          @input="val => emitInput('initialOrganisationDepartmentName', val)" />
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue:58]: This was moved intentionally here

File: client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue [hunk 13]
```diff
@@ -95,6 +77,16 @@
...
+        <dp-input
+          v-if="hasPermission('field_statement_meta_orga_department_name') && !this.localStatement.attributes.isSubmittedByCitizen"
+          id="statementDepartmentName"
+          v-model="localStatement.attributes.initialOrganisationDepartmentName"
+          class="u-mb-0_5"
+          :disabled="isStatementManual ? false : !editable"
+          :label="{
+            text: Translator.trans('department')
+          }"
+          @input="(val) => emitInput('initialOrganisationDepartmentName', val)" />
```
Comment (gruenbergerdemos) [inline client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMeta.vue:89]: Move to top, see comment above

**Review by gruenbergerdemos — APPROVED** [review]
